kratos/includes/kratos_parameters.h

```diff
@@ -549,6 +549,58 @@ class Parameters
 			}
 		}
 	}
+        
+    bool IsEquivalentTo(Parameters& reference) {
+        //Checks if the names and values are the same, no importance to the order. 
+        //Lists have to be ordered, though! Take into account that in Kratos some physical vectors are represented with a list.
+        
+        for (rapidjson::Value::ConstMemberIterator itr = this->mpvalue->MemberBegin(); itr != this->mpvalue->MemberEnd(); ++itr)
+        {
+            std::string item_name = itr->name.GetString();
+            
+            bool found = false;
+            
+            for (rapidjson::Value::ConstMemberIterator itr_ref = reference.mpvalue->MemberBegin(); itr_ref != reference.mpvalue->MemberEnd(); ++itr_ref)
+            {
+                if(item_name == itr_ref->name.GetString()) {
+                    found = true;
+                    Parameters subobject = (*this)[item_name];
+                    Parameters reference_subobject = reference[item_name];
+                    
+                    if(itr->value.IsObject()) {                        
+                        if ( ! subobject.IsEquivalentTo(reference_subobject) ) return false;
+                    }
+                    else {
+                        if(itr->value != itr_ref->value) return false;
```

Comment:
I think the method does what it says (Equivalent), and I would leave it like this. But I agree that it would be useful to add another method (very similar) that only checks the type, after modifications have been done by other processes.